src/features/selection/js/selection.js

```diff
@@ -664,7 +664,7 @@
                   allowCellFocus: true
                 };
 
-                uiGridCtrl.grid.addRowHeaderColumn(selectionRowHeaderDef, 0);
+                uiGridCtrl.grid.addRowHeaderColumn(selectionRowHeaderDef, 0, true);
```

Comment:
I added a new attribute to prevent build columns from being triggered multiple times in a row. ColumnDefs is now responsible for building the columns. See line 763 of [Grid.js](https://github.com/mportuga/ui-grid/blob/75f98c3eda3e46f14f9d924e6f5a6d6e35bcc2b5/src/js/core/factories/Grid.js#L763)